includes/api/save-abandonment-data.php

```diff
@@ -49,114 +49,360 @@ function saveCartAbandonmentData(WP_REST_Request $request)
-            'wcf_shipping_address_2'  => $razorpayData['customer_details']['shipping_address']['line2'] ?? '',
-            'wcf_shipping_city'       => $razorpayData['customer_details']['shipping_address']['city'] ?? '',
-            'wcf_shipping_state'      => $razorpayData['customer_details']['shipping_address']['state'] ?? '',
-            'wcf_shipping_postcode'   => $razorpayData['customer_details']['shipping_address']['zipcode'] ?? '',
-            'wcf_order_comments'      => "",
-            'wcf_first_name'          => $razorpayData['customer_details']['shipping_address']['name'] ?? '',
-            'wcf_last_name'           => "",
-            'wcf_phone_number'        => $razorpayData['customer_details']['shipping_address']['contact'] ?? '',
-            'wcf_location'            => $razorpayData['customer_details']['shipping_address']['country'] ?? '',
-        );
-
-        $checkoutDetails = array(
-            'email'         => $razorpayData['customer_details']['email'] ?? $customerEmail,
-            'cart_contents' => serialize($products),
-            'cart_total'    => sanitize_text_field($cartTotal),
-            'time'          => $currentTime,
-            'other_fields'  => serialize($otherFields),
-            'checkout_id'   => wc_get_page_id('cart'),
-        );
-
-        $sessionId = WC()->session->get('wcf_session_id');
-
-        $cartAbandonmentTable = $wpdb->prefix . "cartflows_ca_cart_abandonment";
-
-        $logObj['checkoutDetails'] = $checkoutDetails;
-
-        if (empty($checkoutDetails) == false) {
-            $result = $wpdb->get_row(
-                $wpdb->prepare('SELECT * FROM `' . $cartAbandonmentTable . '` WHERE session_id = %s and order_status IN (%s, %s)', $sessionId, 'normal', 'abandoned') // phpcs:ignore
-            );
+        // Save Abondonment data for Abondonment cart lite 
+       $result = saveWooAbandonmentCartLiteData($razorpayData, $wcOrderId);
+
+       return new WP_REST_Response($result['response'],  $result['status_code']);
+    }
+    else
+    {
+        $response['status'] = false;
+        $response['message'] = 'Failed to insert data';
+        $statusCode = 400;
+
+        $logObj['response'] = $response;
+        $logObj['status_code'] = $statusCode;
+        rzpLogInfo(json_encode($logObj));
+    }
+    
+
+    return new WP_REST_Response( $response, $statusCode);  
+}
+
+//Save abandonment data for woocommerce Abondonment cart lite plugin
+function saveWooAbandonmentCartLiteData($razorpayData, $wcOrderId)
+{
+    global $woocommerce;
+    global $wpdb;
+    $billingFirstName = $razorpayData['customer_details']['billing_address']['name'] ?? '';
+    $billingLastName = " ";
+    $billingZipcode = $razorpayData['customer_details']['billing_address']['zipcode'] ?? '';
+    $shippingZipcode  = $razorpayData['customer_details']['shipping_address']['zipcode'] ?? '';
+
+    $shippingCharges = WC()->cart->shipping_total;
```

Comment:
@poojarazor Where this cart object initiated? Will it calculate the shipping charges properly?

Reply:
@poojarazor Update on this?